### DISCUSSION TITLE
add mimalloc as optional dependency

### DISCUSSION THREAD

**Discussion by pca006132** [elalish/manifold#719] [Ideas]
mimalloc is faster than glibc on linux, and it is probably also faster than the default allocator on Windows and Mac.
For extreme examples such as `Samples.Sponge4`, the difference can be up to 12%.

While on linux we can use `LD_PRELOAD` to override the default memory allocator, I wonder if we should add mimalloc into our code as an optional dependency. This can allow our pypi build to use mimalloc and benefit from the performance improvement.

**Comment by elalish** [comment]
I don't have a strong opinion. Does it help any other examples much? Does it slow any down?

**Reply by pca006132** [reply]
Yes it helps in general. So far I don't see any slow down, but I have only tested on Linux. Would be helpful if others can test on Mac and Windows. Will propose a patch later for testing.
